### ISSUE TITLE
[REVIEW]: TorchGAN: A Flexible Framework for GAN Training and Evaluation

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2606]
**Submitting author:** @avik-pal (<a href="http://orcid.org/0000-0002-3938-7375">Avik Pal</a>)
**Repository:** <a href="https://github.com/torchgan/torchgan/" target ="_blank">https://github.com/torchgan/torchgan/</a>
**Version:** v0.0.4
**Editor:** @terrytangyuan
**Reviewer:** @Paul92, @urbanophile
**Archive:** Pending

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/78a73cb647a355a8d6826ad2f2086bcc/status.svg)](https://joss.theoj.org/papers/78a73cb647a355a8d6826ad2f2086bcc)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/78a73cb647a355a8d6826ad2f2086bcc"><img src="https://joss.theoj.org/papers/78a73cb647a355a8d6826ad2f2086bcc/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/78a73cb647a355a8d6826ad2f2086bcc/status.svg)](https://joss.theoj.org/papers/78a73cb647a355a8d6826ad2f2086bcc)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/torchgan/torchgan/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@Paul92 & @urbanophile, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @terrytangyuan know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/torchgan/torchgan/">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@avik-pal) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/torchgan/torchgan/">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@avik-pal) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @Paul92, @urbanophile it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2606 with the following error: 

 Can't find any papers to compile :-(

**Comment by avik-pal** [comment]
@whedon generate pdf from branch ap/joss_paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch ap/joss_paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02606/joss.02606/10.21105.joss.02606.pdf)

**Comment by kthyng** [comment]
Hello! I am the rotating associate editor in chief and am checking in on stale reviews. Will reviewers @Paul92, @urbanophile have a chance to start on/continue your reviews sometime soon? It's been over a month since the review started. Thanks!

**Comment by urbanophile** [comment]
Hi apologies, yes. I will continue on the review.

**Comment by avik-pal** [comment]
bump.

**Comment by terrytangyuan** [comment]
@Paul92 and @urbanophile, did you get a chance to review this yet?

**Comment by arfon** [comment]
:wave: @Paul92 and @urbanophile - could you please update us on the status of your reviews?

**Comment by arfon** [comment]
I've just emailed both @Paul92 and @urbanophile to see if they are able to return to this review sometime soon. I'm sorry this is taking so long @avik-pal.

**Comment by arfon** [comment]
@avik-pal - I'm afraid I think we need to find some new reviewers here 😞 . Could you take a look a [this list of potential reviewers](https://bit.ly/joss-reviewers) and identify a few people who would be good candidates to review this submission?

**Comment by arfon** [comment]
@whedon assign me as editor

**Comment by arfon** [comment]
@vs74 @NMontanaBrown - would you be willing to review this submission for JOSS? We carry out our checklist-driven reviews here in GitHub issues and follow these guidelines: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html

The submission we'd be asking you to review is [_TorchGAN: A Flexible Framework for GAN Training and Evaluation_](https://github.com/torchgan/torchgan/)

**Comment by NMontanaBrown** [comment]
Hi @arfon, I should be able to, if this is still in need of a reviewer.

**Comment by arfon** [comment]
> Hi @arfon, I should be able to, if this is still in need of a reviewer.

Yes please! I'll go ahead and set you up on the review now.

**Comment by arfon** [comment]
@whedon add @NMontanaBrown as reviewer

**Comment by whedon** [comment]
OK, @NMontanaBrown is now a reviewer

**Comment by arfon** [comment]
@whedon re-invite @NMontanaBrown as reviewer

**Comment by whedon** [comment]
@nmontanabrown already has access.

**Comment by arfon** [comment]
@NMontanaBrown - Please read the "Reviewer instructions & questions" in the first comment above.

As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `https://github.com/openjournals/joss-reviews/issues/2606` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for the review process to be completed within about 4-6 weeks but please make a start well ahead of this as JOSS reviews are by their nature iterative and any early feedback you may be able to provide to the author will be very helpful in meeting this schedule.

**Comment by arfon** [comment]
:wave: folks. Checking in here to see how things are going. @NMontanaBrown it seems like you made it a good way through your review but might be waiting on @avik-pal to make some updates?

**Comment by arfon** [comment]
@whedon remove @urbanophile as reviewer

**Comment by whedon** [comment]
OK, @urbanophile is no longer a reviewer

**Comment by arfon** [comment]
@whedon remove @Paul92 as reviewer

**Comment by whedon** [comment]
OK, @Paul92 is no longer a reviewer

**Comment by arfon** [comment]
:wave: @thelinuxmaniac @smith42 – would either of you be willing to review this submission for JOSS? We carry out our checklist-driven reviews here in GitHub issues and follow these guidelines: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html

The submission we'd be asking you to review is [_TorchGAN: A Flexible Framework for GAN Training and Evaluation_](https://github.com/torchgan/torchgan/). We already have one review complete and are seeking a second reviewer to wrap up our review process.

**Comment by NMontanaBrown** [comment]
> 👋 folks. Checking in here to see how things are going. @NMontanaBrown it seems like you made it a good way through your review but might be waiting on @avik-pal to make some updates?

I am waiting on the updates regarding unit tests to assert the functionality of the package, raised in this [ticket](https://github.com/torchgan/torchgan/issues/138) before finishing the review.

**Comment by arfon** [comment]
> I am waiting on the updates regarding unit tests to assert the functionality of the package, raised in this ticket before finishing the review.
 
Got it. Thanks for the update! @avik-pal – are you planning on making these changes sometime in the not-too-distant future?

**Comment by avik-pal** [comment]
Yes, I am planning to do it by the end of this month. Apologies for the delay. I was in the middle of graduation and shifting and hence haven't been able to work on the tests.

**Comment by arfon** [comment]
Friendly reminder to get to these changes when you can @avik-pal 😄

**Comment by avik-pal** [comment]
Sorry I should have updated here (I mentioned in the other issue). The tests have been updated

**Comment by NMontanaBrown** [comment]
@avik-pal  thanks, list completed @arfon

**Comment by arfon** [comment]
:wave: @yxoos –  would you be willing to review this submission for JOSS? We carry out our checklist-driven reviews here in GitHub issues and follow these guidelines: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html

We already have one review from @NMontanaBrown and are looking for a second reviewer to help us wrap up the JOSS review process.

The submission we'd be asking you to review is [_TorchGAN: A Flexible Framework for GAN Training and Evaluation_](https://github.com/torchgan/torchgan/). We already have one review complete and are seeking a second reviewer to wrap up our review process.

**Comment by danielskatz** [comment]
@arfon - just a ping on this...

**Comment by arfon** [comment]
I've asked @terrytangyuan if he's willing to provide a second review here. 

@avik-pal – I'd still welcome any help you might be able to provide to identify a second reviewer here.

**Comment by terrytangyuan** [comment]
Yes, I am happy to switch to the reviewer for this submission and my review is actually going to be quick as I initially spent some time looking through this submission as an editor previously.

**Comment by terrytangyuan** [comment]
@whedon add @terrytangyuan as reviewer

**Comment by whedon** [comment]
OK, @terrytangyuan is now a reviewer

**Comment by terrytangyuan** [comment]
## Review checklist for @terrytangyuan
### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/torchgan/torchgan/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@avik-pal) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by terrytangyuan** [comment]
@avik-pal Just some small feedback:
1. There might be some papers that you cited in arXiv but have been published to somewhere more formally. Could you check and update them accordingly?
2. The paper has been here for a year and some things might have been changed, such as the feature tables for existing frameworks. Could you update those wherever necessary? If the API of TorchGAN has changed, please update as well.

**Comment by arfon** [comment]
@avik-pal – how are you getting on with the reviewer feedback here?

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02606/joss.02606/10.21105.joss.02606.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02606/joss.02606/10.21105.joss.02606.pdf) :page_facing_up: :point_left:

**Comment by avik-pal** [comment]
Yes the latest pdf has the updated references

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.5575758 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5575758" target="_blank">10.5281/zenodo.5575758</a> is the archive.

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/cvpr.2016.90 is OK
- 10.1109/cvpr.2017.243 is OK
- 10.1109/iccv.2017.304 is OK
- 10.1109/iccv.2017.244 is OK
- 10.1109/cvpr.2017.19 is OK
- 10.1109/cvpr.2018.00916 is OK

MISSING DOIs

- 10.1109/cvpr.2018.00165 may be a valid DOI for title: MoCoGAN: Decomposing Motion and Content for Video Generation

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@whedon accept deposit=true from branch ap/joss_paper

**Comment by arfon** [comment]
@NMontanaBrown, @terrytangyuan – many thanks for your reviews here! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@avik-pal – your paper is now accepted and published in JOSS :zap::rocket::boom:
